src/shogun/lib/observers/ParameterObserver.h

```diff
@@ -82,12 +83,77 @@ namespace shogun
 		 */
 		virtual bool filter(const std::string& param);
 
+		/**
+		 * Get a vector with all the observations matching a certain name
+		 * and a certain name (not SG_REF).
+		 * @tparam T the type of the observations
+		 * @param name the name of the observations
+		 * @return vector with the observations casted to the requested type
+		 */
+		std::vector<ObservedValue*> get_observations(std::string name)
```

Comment:
who would call this? Swig?

Reply:
Because it will cause problems ... ;)

Reply:
Yeah, this was a draft of a possible design for getting the observations. I thought that SWIG would have caused problems ;)

Reply:
swig should use the new `get(name, index)`.
For that `ObservedValue` will need to added to the get dispatcher (you already added `CMachine`)

Reply:
Is this even needed?